src/test/kotlin/io/imotions/bson4k/decoder/BsonPrimitivesDecoderTest.kt

```diff
@@ -168,4 +182,21 @@ class BsonPrimitivesDecoderTest : StringSpec({
             bson.decodeFromBsonDocument<String>(Document().toBsonDocument())
         }
     }
+
+    "Throw exception if implicit conversion from INT64 to INT32 is out-of-bounds" {
+        shouldThrow<BsonDecodingException> {
+            val document = Document(VALUE_KEY, Long.MAX_VALUE).toBsonDocument()
```

Comment:
Minor: `Long.MAX_VALUE` confused me at first; basically any value > `Int.MAX_VALUE` would work, right? `Int.MAX_VALUE + 1` may be more expressive.